device/tt_cluster_descriptor_types.h

```diff
@@ -4,24 +4,63 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#pragma once 
+#pragma once
 
-#include <tuple>
+#include <fmt/format.h>
 
-using chip_id_t = int;
-using ethernet_channel_t = int;
-using eth_coord_t = std::tuple<int, int, int, int>;  // x, y, rack, shelf
+#include <functional>
+#include <type_traits>
+
+
+namespace tt::umd {
+
+enum class chip_id : int {
+    none = -1,
+};
+
+enum class ethernet_channel : int {};
+
+struct eth_coord {
+    int x;
+    int y;
+    int rack;
+    int shelf;
+
+    // in C++20 this should be defined as:
+    // constexpr bool operator==(const eth_coord &other) const noexcept = default;
+    constexpr bool operator==(const eth_coord &other) const noexcept {
+        return (x == other.x and y == other.y and rack == other.rack and shelf == other.shelf);
+    }
+};
+
+}  // namespace tt::umd
+
+using chip_id_t = tt::umd::chip_id;
+using ethernet_channel_t = tt::umd::ethernet_channel;
+using eth_coord_t = tt::umd::eth_coord;
```

Comment:
if this is only merged to `metal-main` then you can potentially get rid of these `*_t`?